Lib/unittest/mock.py

```diff
@@ -2222,10 +2222,11 @@ def __init__(self, name, parent):
     def create_mock(self):
         entry = self.name
         parent = self.parent
-        m = parent._get_child_mock(name=entry, _new_name=entry,
-                                   _new_parent=parent)
-        setattr(parent, entry, m)
-        _set_return_value(parent, m, entry)
+        with NonCallableMock._lock:
```

Comment:
Does this lock already exist? I can't see where it's been added?

Reply:
Yes, `NonCallableMock._lock` does already exist:
https://github.com/python/cpython/blob/f5a9c34f38886c5cf9c2f8d860eee3473447e030/Lib/unittest/mock.py#L442